In [10]:
# Load all the libraries needed for running this notebook
library(tidyverse)
library(repr)
options(repr.matrix.max.rows = 4) #limits output of dataframes to 6 rows

# Data Description
## Players Dataset
Run the code below to load players.csv into a tibble named `players`.  

In [11]:
# load the players dataset from GitHub into an R dataframe (tibble). 
players_url <- "https://raw.githubusercontent.com/Aylin-Ab/dsci-100-2025w1-group-27/refs/heads/main/players.csv"
players <- read_csv(players_url)
players

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1   Pro        TRUE     
2   Veteran    TRUE     
⋮   ⋮          ⋮        
195 Amateur    FALSE    
196 Pro         TRUE    
    hashedEmail                                                     
1   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
2   f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9
⋮   ⋮                                                               
195 f19e136ddde68f365afc860c725ccff54307dedd13968e896a9f890c40aea436
196 d9473710057f7d42f36570f0be83817a4eea614029ff90cf50d8889cdd729d11
    played_hours name      gender Age
1   30.3         Morgan    Male    9 
2    3.8         Christian Male   17 
⋮   ⋮            ⋮         ⋮      ⋮  
195 2.3          Harlow    Male   17 
196 0.2          Ahmed     Other  NA

This data frame includes a list of all unique players, including data about each player. After loading the dataset, you can see that the `players` tibble consists of **7 variables** (columns) and **196 observations** (rows).  

  Here's what each variable in the dataset means:
  - **`experience`**: player's experience level, chosen from a menu with 5 different levels (see all experience levels in the output of below code)
  - **`subscribe`**: whether the player subscribed to a game-related newsletter (TRUE = subscribed, FALSE = didn't subscribe)
  - **`hashedEmail`**: an anonymized version of each player's email address used to link player data across the two datasets
  - **`played_hours`**: total number of hours, across different sessions, the player has spent playing on the server
  - **`name`**: player's chosen display name/alias
  - **`gender`** player's gender, chosen from a menu with 7 different genders (see all gender levels in the output of below code)
  - **`Age`**: player's age in years

In [12]:
# extract the experience column and find all unique experience levels
experience_levels <- players |>
    pull(experience) |>
    unique()

# extract the gender column and find all unique gender levels
gender_levels <- players |>
    pull(gender) |>
    unique()

experience_levels
gender_levels

[1] "Pro"      "Veteran"  "Amateur"  "Regular"  "Beginner"

[1] "Male"              "Female"            "Non-binary"       
[4] "Prefer not to say" "Agender"           "Two-Spirited"     
[7] "Other"

The table below summarizes each variable in the players dataset, showing its original and suggested data type, and whether it is categorical or numerical.

The `experience` and `gender` variables are stored as *character* by default, but since they have a few distinct levels, they can be treated as **factors** for analysis. Although the choice of data type ultimately depends on the research question. Variables marked N/A already have the correct data type.

| Variable Name | Original Data Type | Suggested Data Type | Categorical vs Numerical |
| ------------- | ---------------    | --------------------| -----------------------  |
| experience    | chr (character)    | fct (factor)        |  Categorical             |
| subscribe     | lgl (logical)      | N/A                 |  *Categorical (Logical)  |
| hashedEmail   | chr (character)    | N/A                 |  Categorical             |
| played_hours  | dbl (double)       | N/A                 |  Numerical               |
| name          | chr (character)    | N/A                 |   Categorical            |
| gender        | chr (character)    | fct (factor)        |   Categorical            |
| Age           | dbl (double)       | N/A                 |  Numerical               |